service/src/main/java/uk/gov/hmcts/reform/fpl/service/orders/history/SealedOrderHistoryService.java

```diff
@@ -103,10 +112,33 @@ public Map<String, Object> generate(CaseData caseData) {
     }
 
     public GeneratedOrder lastGeneratedOrder(CaseData caseData) {
+        return lastGeneratedOrderElement(caseData).getValue();
+    }
+
+    private Element<GeneratedOrder> lastGeneratedOrderElement(CaseData caseData) {
         return caseData.getOrderCollection().stream()
             .min(legacyLastAndThenByDateAndTimeIssuedDesc())
-            .orElseThrow(() -> new IllegalStateException("Element not present"))
-            .getValue();
+            .orElseThrow(() -> new IllegalStateException("Element not present"));
+    }
+
+    public Map<String, Object> processUploadedOrder(CaseData caseData) {
+        Map<String, Object> data = new HashMap<>();
+        // get the latest order just created in about-to-submit callback;
+        Element<GeneratedOrder> orderElm = lastGeneratedOrderElement(caseData);
+        GeneratedOrder order = lastGeneratedOrder(caseData);
```

Comment:
can we not just call orderElm.getValue() to skip doing the call to `lastGeneratedOrderElement` twice?

Reply:
yeah right! updated it.